http/http-advanced/src/main/java/io/quarkus/ts/http/advanced/HeadersMessageBodyWriter.java

```diff
@@ -0,0 +1,28 @@
+package io.quarkus.ts.http.advanced;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+
+import jakarta.ws.rs.WebApplicationException;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.MultivaluedMap;
+import jakarta.ws.rs.ext.MessageBodyWriter;
+import jakarta.ws.rs.ext.Provider;
+
+@Provider
+public class HeadersMessageBodyWriter implements MessageBodyWriter<String> {
+
+    @Override
+    public boolean isWriteable(Class aClass, Type type, Annotation[] annotations, MediaType mediaType) {
+        return String.class.isAssignableFrom(aClass) && MediaType.TEXT_PLAIN_TYPE.isCompatible(mediaType);
+    }
+
+    @Override
+    public void writeTo(String s, Class aClass, Type type, Annotation[] annotations, MediaType mediaType,
+            MultivaluedMap multivaluedMap, OutputStream outputStream) throws IOException, WebApplicationException {
+        final var content = "Headers response: " + s;
```

Comment:
I think this can be `String`

Reply:
what is wrong with `var`?

Reply:
I mean, it exists to cut verbosity/unnecessary info where the type is obvious (AFAICT, no idea what is official reason), and here, you know :-)

Reply:
though I don't think @jcarranzan cares about that :-D it seems like this is more or less c&p for https://github.com/quarkusio/quarkus/pull/41411/files :-/

Reply:
> what is wrong with var?

I would say nothing just that I would use that in more complicated assigment of variable (like concatenating multiple strings together) or in foreach where I don't care that much about type. 

Anyway I'll leave the decision to @jcarranzan what he want to use.

Reply:
IMHO, var or String here it's irrelevant, at least if readability get worse, I would use strong typed as String.
Well @michalvavrik u are wrong in this case, I didn't realize about that code  and upstream would be a c&p from https://github.com/jholleran/message-body-writer-selection and so on...

Reply:
> I would say nothing just that I would use that in more complicated assigment of variable (like concatenating multiple strings together) or in foreach where I don't care that much about type.

Ok, I don't care, I was just curious as I use it all the time.

Reply:
> Well @michalvavrik u are wrong in this case, I didn't realize about that code and upstream would be a c&p from https://github.com/jholleran/message-body-writer-selection and so on...

got it

Reply:
btw

> concatenating multiple strings together

=> always string so `String` doesn't provide me a new info :-)